src/Glpi/Controller/DropdownFormController.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Controller;
+
+use CommonDevice;
+use CommonDropdown;
+use Html;
+use Glpi\Event;
+use Symfony\Component\HttpFoundation\Exception\BadRequestException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\Routing\Attribute\Route;
+use Toolbox;
+
+final class DropdownFormController extends AbstractController
+{
+    #[Route("/dropdown/form/{class}", name: "glpi_dropdown_form")]
+    public function __invoke(Request $request): Response
+    {
+        $class = $request->attributes->getString('class');
+
+        if (!$class) {
+            throw new BadRequestException('The "class" attribute is mandatory for dropdown form routes.');
+        }
+
+        if (!\is_subclass_of($class, CommonDropdown::class)) {
+            throw new BadRequestException('The "class" attribute must be a valid dropdown class.');
+        }
+
+        return new StreamedResponse(function () use ($class, $request) {
+            $dropdown = new $class();
+            $this->loadDropdownForm($request, $dropdown);
+        });
+    }
+
+    public static function loadDropdownForm(Request $request, CommonDropdown $dropdown, ?array $options = null): void
```

Comment:
Are we obligated to use `$_POST` and `$_GET` in this method ? Can't we get the information from `$request` ?

Reply:
Indeed, I will fix this.